pages/plants/[id]/edit.js

```diff
@@ -0,0 +1,29 @@
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Form from "@/components/Form";
+
+export default function EditPage({ plants, handleEditPlant }) {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const editPlant = plants.find((plant) => plant.id === id);
+
+  if (!editPlant) return null;
```

Comment:
An early return with an output for the user would be interesting here, e.g. that the ID does not exist with a return link.